src/scm/github.js

```diff
@@ -0,0 +1,433 @@
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      })
+      .then((response) => {
+        return {file: file, blob: response};
+      })
+    });
+    if (changed.length === 0) {
+      showAlert('Nothing to do', LEVEL_WARN);
+      return;
+    }
+
+    Promise.all([
+      Promise.all(promises),
+      $.getJSON(
+        `${this.baseUrl}/repos/${context.repo.fullName}/branches/${context.branch}`,
+        { access_token: this.accessToken }
+      )
+    ])
+    .then((responses) => {
+      return $.getJSON(
+        responses[1].commit.commit.tree.url,
+        { 
+          recursive: 1,
+          access_token: this.accessToken 
+        }
+      )
+      .then((baseTree) => {
+        const tree = responses[0].map((data) => {
+          return {
+            path: data.file,
+            mode: '100644',
+            type: 'blob',
+            sha: data.blob.sha
+          }
+        })
+        .concat(baseTree.tree.filter((t) =>  {
+          return (t.type != 'tree') && (!/.(gs|html)$/.test(t.path) || unchanged.indexOf(t.path) >= 0);
+        }));
+        return {
+          tree: tree
+        };
+      })
+      .then((payload) => {
+        return $.ajax({
+          url: `${this.baseUrl}/repos/${context.repo.fullName}/git/trees`,
+          headers: {
+            'Authorization': `token ${this.accessToken}`
+          },
+          method: 'POST',
+          crossDomain: true,
+          dataType: 'json',
+          contentType: 'application/json',
+          data: JSON.stringify(payload)
+        });
+      })
+      .then((response) => {
+        return Object.assign(response, { parent: responses[1].commit.sha })
+      })
+      .fail((err) => {
+        throw err;
+      })
+    })
+    .then((response) => {
+      const payload = {
+        message: $('#commit-comment').val(),
+        tree: response.sha,
+        parents: [
+          response.parent
+        ]
+      };
+      return $.ajax({
+        url: `${this.baseUrl}/repos/${context.repo.fullName}/git/commits`,
+        headers: {
+          'Authorization': `token ${this.accessToken}`
+        },
+        method: 'POST',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      });
+    })
+    .then((response) => {
+      const payload = {
+        force: true,
+        sha: response.sha
+      };
+      return $.ajax({
+        url: `${this.baseUrl}/repos/${context.repo.fullName}/git/refs/heads/${context.branch}`,
+        headers: {
+          'Authorization': `token ${this.accessToken}`
+        },
+        method: 'PATCH',
+        crossDomain: true,
+        dataType: 'json',
+        contentType: 'application/json',
+        data: JSON.stringify(payload)
+      });
+    })
+    .then(() => {
+      showAlert(`Successfully push to ${context.branch} of ${context.repo.name}`);
+    })
+    .catch((err) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/65105829498a48107e9a8b01cb705b310fffdc68/src/scm/github.js#L134"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>